src/content_stream.cpp

```diff
@@ -43,118 +43,164 @@ void ContentReader::close() {
   }
 }
 
-std::string ContentReader::read() {
-  matched_idx = 0;
+// Limit message sizes to 10 MB to prevent DoS issues
+#define MAX_CONTENT_LENGTH (100000000)
 
-  // Find Content-Length header prefix
-  if (!scan("Content-Length:")) {
-    return "";
+std::string ContentReader::read() {
+  // Currently, a DAP message is encoded like:
+  // Content-Length: 2\r\n
+  // \r\n
+  // {}
+  auto contentLengthResult = readContentLength();
+  if (!contentLengthResult.first) {
+    return invalidMsg();
```

Comment:
The reason I went to the effort of ignoring non DAP messages is that DAP is often used on a stdout / stdin pipe. It's all too common for applications to `printf` some logging for initialization, in debug, or because of some third-party library. From a purest perspective, I understand why you'd sever the connection when something unexpected appears, but unfortunately is likely to break existing code.

Reply:
Good to know! This change is definitely incorrect then.